### ISSUE TITLE
textsum: creating original test data

### ISSUE THREAD

**Issue by mettinger** [tensorflow/models#346]
Hello,

I've successfully trained and tested the textsum model on the toy data (data/data) and would like to test the model on my own data.  There is a very brief comment in the DataSet section about gleaning data format information from the ExampleGen function in data.py but I'm still puzzled.  Is there a straightforward way to convert ordinary text files to binary files properly formatted for input to the model?  

Thanks!

**Comment by prb12** [comment]
Please note: general support questions should really be posted to StackOverflow.  This issues tracker is for code bugs/feature requests.

**Comment by peterwilli** [comment]
@mettinger how did you used te toy data? I couldn't go any further at the 'rename files' part, I renamed data training-0 but I didnt know where to get validation-0 from. Thanks!

**Comment by mettinger** [comment]
@peterwilli Here are the two commands I used for training and testing.  Notice that I tested on the training set, which is not the proper thing to do but I wanted to get things working.

bazel-bin/textsum/seq2seq_attention --mode=train --article_key=article --abstract_key=abstract  --data_path=textsum/data/data --vocab_path=textsum/data/vocab --log_root=textsum/log_root --train_dir=textsum/log_root/train

bazel-bin/textsum/seq2seq_attention   --mode=decode   --article_key=article   --abstract_key=abstract   --data_path=textsum/data/data   --vocab_path=textsum/data/vocab   --log_root=textsum/log_root   --decode_dir=textsum/log_root/decode   --beam_size=8

**Comment by peterwilli** [comment]
@mettinger thanks, I thought you had to run all 3 commands. I'm new to TF but I'm trying to get it up and running to support my research. Thanks a lot! Will try it tonight

**Comment by panyx0718** [comment]
See https://github.com/tensorflow/models/pull/379/files for examples of making training data for the model.

**Comment by tumusudheer** [comment]
@mettinger .. I'm  using 
bazel-bin/textsum/seq2seq_attention --mode=train --article_key=article --abstract_key=abstract --data_path=textsum/data/data --vocab_path=textsum/data/vocab --log_root=textsum/log_root --train_dir=textsum/log_root/train
to train the model with toy data. But the model is not converging and kept running. How long does your training ran with the toy data ? Do you have to stop it manually ? How do we know the model got converged.

All I can see on the terminal is avg_loss as below:
running_avg_loss: 0.917912
running_avg_loss: 0.967694
running_avg_loss: 0.949833
running_avg_loss: 0.832837
running_avg_loss: 1.041382
running_avg_loss: 1.417824
running_avg_loss: 1.231523
running_avg_loss: 1.252483
running_avg_loss: 0.943122
running_avg_loss: 1.17804

It kept running for more than 12 hrs on my machine with Nvidia GPU GEFORCE GTX version 9X. To get the model converged,do we have to change any params in seq2seq_attention.py or any other python files?

**Comment by monajalal** [comment]
@tumusudheer did you get any response? Did your training using the toy data ever converge? How long did it take? How much RAM does your GPU have?

**Comment by licangqiong** [comment]
@tumusudheer  I do have the same problem. I saw the default max step to stop is 1000000 steps.
It kept running for more than 24 hrs using the toy data, which contains only 21 examples.

**Comment by panyx0718** [comment]
There is no need to train the toy data for that long. It's tiny and will overfit. real dataset doesn't go to <1 loss

**Comment by xtr33me** [comment]
@panyx0718 Thanks for this information as I am currently going through training on my own data and this will be helpful as I was expecting that I would have to wait until the loss was <1.

**Comment by prakhar21** [comment]
@mettinger Hi, I have also trained the model on the sample data, already provided. But now if I run with the decode parameter on the same training data. The decoder file inside log_root/decoder does not contain any text other that <UNK>

**Comment by xtr33me** [comment]
Please refer to the issue #464

You are using the included vocab file but that does not contain the words that are in the training data. In the ticket included above, I have referenced a modified version of the vocab file that you will find a link for and if you train against that you will atleast see more than UNK.

That said, it is very important to know that you will still not get good results unless you train against 1 million+ articles. I scraped my own 40k articles and the results still weren't good, however when I brought it up to over 1 million I started seeing pretty good results. I believe the reasoning for this is due to the fact that this is an abstractive model, which lends itself to needing a lot of data.

**Comment by vgoklani** [comment]
I just wanted to follow-up on this thread:

1. @mettinger were you able to add new data for testing? Were you able to decipher the input schema?
2. @tumusudheer I also ran into the same issue regarding over-fitting of the example data. Did you find a way to enable an "early-stopping" criteria? 

Thanks :)

**Comment by xtr33me** [comment]
@vgoklani Just reference the link @panyx0718 provided above. It references data_convert_example which will provide you all you need to decode the sample data and provide the schema.  Also as far overfitting, just break out by using ctrl-c.  

I believe Xin made a statement in another ticket that you should normally not go below and avg loss of 1 or you will find yourself overfitting. Now of course this is a rough estimate as it is also based on the dataset used. With a little bit over 1.3 million, I found the model starting to overfit around 1.8ish.  

Now please keep in mind I am learning this as I go, but from reading I found that I was able to see when the model was starting to overfit by using a combo of the avg loss chart in tensorboard and eval on another dataset.  When I saw that the avg loss was getting far below what the model was doing with my run of eval, I stopped training.

Again, take this all with a grain of salt as it is just based on my deductions and I have not gotten any validation as to whether this is correct, but it has seemed to work well.  Hope it helps!

**Comment by licangqiong** [comment]
@xtr33me  Is it possible for you to opensource the scrap code and scrapped data? Thank you.

**Comment by brucedeng** [comment]
@xtr33me  do you have any benchmark on your training timing on # of gpus or cpus for 40K or 1M articles? would like to see any results on that. thank you!

**Comment by gunan** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
